backend/service/authn/oidc.go

```diff
@@ -296,6 +296,64 @@ func DefaultClaimsFromOIDCToken(ctx context.Context, t *oidc.IDToken) (*Claims,
 	}, nil
 }
 
+func NewClaimsConfig(subjectClaim, groupsClaim string) *ClaimsConfig {
+	return &ClaimsConfig{
+		subjectClaim: subjectClaim,
+		groupsClaim:  groupsClaim,
+	}
+}
+
+type ClaimsConfig struct {
+	subjectClaim string
+	groupsClaim  string
+}
+
+func (cc *ClaimsConfig) ClaimsFromOIDCToken(ctx context.Context, t *oidc.IDToken) (*Claims, error) {
+	claims := make(map[string]interface{})
+	if err := t.Claims(&claims); err != nil {
+		return nil, err
+	}
+
+	subjectInt, ok := claims[cc.subjectClaim]
+	if !ok {
+		return nil, fmt.Errorf("claims did not deserialize with %s field", cc.subjectClaim)
+	}
+	subject, ok := subjectInt.(string)
+	if !ok {
+		return nil, fmt.Errorf("claims did not deserialize with %s field", cc.subjectClaim)
+	}
+	if subject == "" {
+		return nil, errors.New("claims did not deserialize with desired fields")
+	}
+	var groups []string
+	if cc.groupsClaim != "" {
+		groupsInt, ok := claims[cc.groupsClaim]
+		if !ok {
+			return nil, fmt.Errorf("claims did not deserialize with %s field", cc.groupsClaim)
+		}
+		groupsIntSlice, ok := groupsInt.([]interface{})
+		if !ok {
+			return nil, fmt.Errorf("group claim did not deserialize with %s field, groups claim must be a list", cc.groupsClaim)
+		}
+
+		for _, v := range groupsIntSlice {
+			group, ok := v.(string)
```

Comment:
I'm curious how you're currently planning on using this? I'm a bit worried about casting this to a string instead of using a specific provider group type (e.g. an Okta group).

Reply:
I feel like if we follow the Kubernetes approach we can assume that the groups claim is a slice of strings:

![image](https://user-images.githubusercontent.com/22289110/170652064-d2344347-bb45-4643-a890-9486b8ad5a49.png)

https://kubernetes.io/docs/reference/access-authn-authz/authentication/#configuring-the-api-server 

I personally,